### ISSUE TITLE
(release/0.5) ./install_requirements.sh failing due to conflicting dependencies, with a haphazard resolution (only for inference)

### ISSUE THREAD

**Issue by himanshunaidu** [pytorch/executorch#10151]
### 🐛 Describe the bug

Greetings, hope you're doing well.
I have been trying to install the Executorch environment on my Mac M3 Sonoma 14.6 using the following environment:
`./install_requirements.sh --pybind coreml mps`

I was facing a bunch of issues but I finally got the environment installed, but I am still determining if this is valid or not.   
Will update the issue as and when I make progress on it. 

Similar to the following issue, I had been getting a dependency issue
https://github.com/pytorch/executorch/issues/9307

```python
The conflict is caused by:
    The user requested torch==2.6.0
    torchvision 0.21.0 depends on torch==2.6.0
    timm 1.0.7 depends on torch
    torchaudio 2.6.0 depends on torch==2.7.0
```

I managed to fix it with a series of actions that I took after analyzing the error logs, to fix the environment. But I am not sure if this still works as expected, and I am still testing this. Here is what I did:
1. Create a new environment with Python=3.10.0

2. Change the torch URL and run

```python
# TORCH_URL = "https://download.pytorch.org/whl/test/cpu"
TORCH_URL = "https://download.pytorch.org/whl/cpu"
```

This fixes the dependency conflicts, and installs the base packages but it fails to build the pip wheel for ExecuTorch, due to some CMake issues. 

3. Downgrade the cmake version to 3.25.0 just as present in the new torch url and rerunNow this where it gets unintuitive. 
I do not clean the environment, and using the base packages (along with downgraded cmake), I rerun the installation script.

This ends up fixing the issue. 

The weird part is the installation ends up upgrading the cmake anyway, so I don't know why downgrading the cmake version worked.  

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: macOS 14.6 (arm64)
GCC version: Could not collect
Clang version: 16.0.0 (clang-1600.0.26.4)
CMake version: Could not collect
Libc version: N/A

Python version: 3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:54:21) [Clang 16.0.6 ] (64-bit runtime)
Python platform: macOS-14.6-arm64-arm-64bit
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Apple M3

Versions of relevant libraries:
[pip3] flake8==7.0.0
[pip3] mypy==1.11.2
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] numpydoc==1.7.0
[conda] numpy                     1.26.4          py312h7f4fdc5_0  
[conda] numpy-base                1.26.4          py312he047099_0  
[conda] numpydoc                  1.7.0           py312hca03da5_0  

cc @larryliu0820 @jathu

**Comment by himanshunaidu** [comment]
This is the cmake error I was getting before downgrading the version and re-running it:
https://gist.github.com/himanshunaidu/8cc557347b1d2d925900bdbab75f2d3d

**Comment by himanshunaidu** [comment]
As it turns out, there are still some packages that end up not being installed, such as coremltools, triton as well as local packages such as executorchcoreml.
So while inference seems to be working, model conversion does not. So this hasn't really been properly fixed.

**May I know in what circumstances do such installation issues occur, where the dependencies are installed, executorch wheel is built, but apple-specific packages are not still installed.**

**Comment by himanshunaidu** [comment]
This is the environment that gets created (I also installed opencv-python, skimage, sklearn, and coremltools to conduct exp):

# packages in environment at /opt/anaconda3/envs/executorch_0.5:
#
# Name                    Version                   Build  Channel
attrs                     25.3.0                   pypi_0    pypi
blas                      1.0                    openblas  
bzip2                     1.0.8                h80987f9_6  
c-ares                    1.34.5               h5505292_0    conda-forge
ca-certificates           2025.2.25            hca03da5_0  
cattrs                    24.1.3                   pypi_0    pypi
certifi                   2025.1.31                pypi_0    pypi
charset-normalizer        3.4.1                    pypi_0    pypi
cmake                     4.0.0                    pypi_0    pypi
coremltools               8.2                      pypi_0    pypi
exceptiongroup            1.2.2                    pypi_0    pypi
execnet                   2.1.1                    pypi_0    pypi
executorch                0.5.0a0+1bc0699          pypi_0    pypi
expat                     2.7.0                h286801f_0    conda-forge
expecttest                0.3.0                    pypi_0    pypi
filelock                  3.18.0                   pypi_0    pypi
flatbuffers               25.2.10                  pypi_0    pypi
freetype                  2.13.3               h1d14073_0    conda-forge
fsspec                    2025.3.2                 pypi_0    pypi
huggingface-hub           0.30.2                   pypi_0    pypi
hypothesis                6.131.0                  pypi_0    pypi
idna                      3.10                     pypi_0    pypi
imageio                   2.37.0          py310hca03da5_0  
iniconfig                 2.1.0                    pypi_0    pypi
jinja2                    3.1.6                    pypi_0    pypi
joblib                    1.4.2           py310hca03da5_0  
krb5                      1.21.3               h237132a_0    conda-forge
lazy_loader               0.4             py310hca03da5_0  
lcms2                     2.17                 h7eeda09_0    conda-forge
lerc                      4.0.0                h313beb8_0  
libcurl                   8.13.0               h73640d1_0    conda-forge
libcxx                    20.1.2               ha82da77_0    conda-forge
libdeflate                1.23                 hec38601_0    conda-forge
libedit                   3.1.20250104    pl5321hafb1f1b_0    conda-forge
libev                     4.33                 h93a5062_2    conda-forge
libexpat                  2.7.0                h286801f_0    conda-forge
libffi                    3.4.4                hca03da5_1  
libgfortran               5.0.0           11_3_0_hca03da5_28  
libgfortran5              11.3.0              h009349e_28  
libjpeg-turbo             3.0.3                h80987f9_0  
liblzma                   5.8.1                h39f12f2_0    conda-forge
liblzma-devel             5.8.1                h39f12f2_0    conda-forge
libnghttp2                1.64.0               h6d7220d_0    conda-forge
libopenblas               0.3.21               h269037a_0  
libpng                    1.6.47               h3783ad8_0    conda-forge
libsqlite                 3.49.1               h3f77e49_2    conda-forge
libssh2                   1.11.1               h9cc3647_0    conda-forge
libtiff                   4.7.0                h551f018_3    conda-forge
libuv                     1.50.0               h5505292_0    conda-forge
libwebp-base              1.5.0                h2471fea_0    conda-forge
libxcb                    1.17.0               hdb1d25a_0    conda-forge
libzlib                   1.3.1                h8359307_2    conda-forge
llvm-openmp               14.0.6               hc6e5704_0  
markupsafe                3.0.2                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
ncurses                   6.5                  h5e97a16_3    conda-forge
networkx                  3.4.2           py310hca03da5_0  
numpy                     2.0.0                    pypi_0    pypi
numpy-base                2.0.1           py310ha9811e2_1  
opencv-python             4.11.0.86                pypi_0    pypi
openjpeg                  2.5.3                h8a3d83b_0    conda-forge
openssl                   3.5.0                h81ee809_0    conda-forge
packaging                 24.2            py310hca03da5_0  
pandas                    2.2.2                    pypi_0    pypi
parameterized             0.9.0                    pypi_0    pypi
pillow                    11.2.1                   pypi_0    pypi
pip                       25.0            py310hca03da5_0  
pluggy                    1.5.0                    pypi_0    pypi
protobuf                  6.30.2                   pypi_0    pypi
pthread-stubs             0.3                  h1a28f6b_1  
pyaml                     25.1.0                   pypi_0    pypi
pytest                    8.3.5                    pypi_0    pypi
pytest-xdist              3.6.1                    pypi_0    pypi
python                    3.10.17         h6cefb37_0_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.10                    6_cp310    conda-forge
pytz                      2025.2                   pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
readline                  8.2                  h1a28f6b_0  
regex                     2024.11.6                pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
rhash                     1.4.3                hb547adb_2    conda-forge
ruamel-yaml               0.18.10                  pypi_0    pypi
ruamel-yaml-clib          0.2.12                   pypi_0    pypi
safetensors               0.5.3                    pypi_0    pypi
scikit-image              0.25.0          py310h313beb8_0  
scikit-learn              1.6.1           py310h313beb8_0  
scipy                     1.15.2          py310h14fe6a6_1  
setuptools                75.8.0          py310hca03da5_0  
six                       1.17.0                   pypi_0    pypi
sortedcontainers          2.4.0                    pypi_0    pypi
sqlite                    3.49.1               hd7222ec_2    conda-forge
sympy                     1.13.1                   pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
threadpoolctl             3.5.0           py310h33ce5c2_0  
tifffile                  2025.2.18       py310hca03da5_0  
timm                      1.0.7                    pypi_0    pypi
tk                        8.6.13               h5083fa2_1    conda-forge
tokenizers                0.21.1                   pypi_0    pypi
tomli                     2.2.1                    pypi_0    pypi
torch                     2.6.0                    pypi_0    pypi
torchao                   0.8.0+gitebc43034          pypi_0    pypi
torchaudio                2.6.0                    pypi_0    pypi
torchsr                   1.0.4                    pypi_0    pypi
torchvision               0.21.0                   pypi_0    pypi
tqdm                      4.67.1                   pypi_0    pypi
transformers              4.47.1                   pypi_0    pypi
typing-extensions         4.13.2                   pypi_0    pypi
tzdata                    2025.2                   pypi_0    pypi
urllib3                   2.4.0                    pypi_0    pypi
wheel                     0.45.1          py310hca03da5_0  
xorg-libxau               1.0.12               h5505292_0    conda-forge
xorg-libxdmcp             1.1.5                hd74edd7_0    conda-forge
xz                        5.8.1                h9a6d368_0    conda-forge
xz-gpl-tools              5.8.1                h9a6d368_0    conda-forge
xz-tools                  5.8.1                h39f12f2_0    conda-forge
zlib                      1.3.1                h8359307_2    conda-forge
zstd                      1.5.6.7                  pypi_0    pypi

**Comment by GregoryComer** [comment]
Thanks for the detailed report. @jathu and @larryliu0820 do you have any thoughts on how we can resolve this?

**Comment by larryliu0820** [comment]
@jathu can you take a look?

**Comment by jathu** [comment]
@himanshunaidu v0.5 does not support CoreML out of the box, and requires you to manually install the dependencies: https://github.com/pytorch/executorch/blob/release/0.5/backends/apple/coreml/scripts/install_requirements.sh.

We've since fixed this and support CoreML out of the box in v0.6. This release also includes fixed to conflicting torch dependencies. If you get a chance, can you try [RC v0.6](https://github.com/pytorch/executorch/tree/release/0.6)? Note this is still a release candidate and has not been officially released

**Comment by himanshunaidu** [comment]
Oh I see. Got it! Thank you so much.

Let me try v0.6 and get back to you.

For additional context:
Before (re)trying the 0.5 version, I had actually already tried the `viable/strict` branch based on instructions here:
https://pytorch.org/executorch/main/using-executorch-building-from-source.html

While it had worked for both conversion and inference, it ended up giving a lot of warnings during conversion.
I tried working with a semantic segmentation model (BiSeNetv2 based on code here: https://github.com/CoinCheung/BiSeNet), by converting it to a `.pte` format, and then assessing its performance.
The performance had a significant drop (mIoU scores dropped from 0.58 to 0.30). 

Hence I have been trying this 0.5 version.
Let me see if the 0.6 version works. If it works but still has the performance drop, I'll close this issue and create a new one. 

Thanks again!

**Comment by himanshunaidu** [comment]
@jathu  
v0.6 is working as you mentioned.
However, the performance drop and warnings are still present exactly as in main branch. 

I will raise a new issue for that. We can close this one.
Thank you!

**Comment by jathu** [comment]
Nice! Yes, please create another issue for the performance bug. Thank you!
